### PR TITLE
Dry run mode

### PR DESCRIPTION
**Pull request by 0xgleb**
## Summary by CodeRabbit

- New Features
  - Added dry-run mode to safely simulate orders without executing them.
  - Introduced a broker-based execution and status polling path for consistent behavior across the app.
- Improvements
  - More reliable order placement with bounded retries and clearer error reporting.
  - Background workers updated to use the broker for execution and status checks, improving stability.
- Tests
  - Substantial new test coverage for on-chain trade parsing, edge cases, and price/share calculations to prevent overexecution and symbol-mapping errors.

### CODE REVIEW

**Comment by 0xgleb** [comment]
@coderabbitai review

**Comment by 0xgleb** [comment]
@coderabbitai review

File: src/schwab/order_poller.rs [hunk 1]
```diff
@@ -146,7 +151,10 @@ impl OrderStatusPoller {
             }
         };
 
-        let order_status = Order::get_order_status(&order_id, &self.env, &self.pool).await?;
+        let order_status = self
+            .broker
+            .get_order_status(&order_id, &self.env, &self.pool)
+            .await?;
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply src/schwab/order_poller.rs:156]: this comment looks out of date

File: src/schwab/order_poller.rs [hunk 2]
```diff
@@ -147,14 +151,16 @@ impl OrderStatusPoller {
...
         if order_status.is_filled() {
             self.handle_filled_order(execution_id, &order_status)
                 .await?;
         } else if order_status.is_terminal_failure() {
-            self.handle_failed_order(execution_id, &order_status)
-                .await?;
+            self.handle_failed_order(execution, &order_status).await?;
         } else {
             debug!(
                 "Order {order_id} (execution {execution_id}) still pending with state: {:?}",
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply src/schwab/order_poller.rs:165]: All checks pass now, so this isn't an issue

**Review by findolor — APPROVED** [review]

**Review by hardyjosh — APPROVED** [review]
